ballet/update.py

```diff
@@ -0,0 +1,126 @@
+        # we're missing keys, figure out which and prompt
+        logger.exception(
+            'Could not create a updated project copy, update failed.')
+    finally:
+        # put back the old replay, if it's been overwritten
+        if old_context is not None:
+            with open(REPLAY_PATH, 'w') as replay_file:
+                json.dump(old_context, replay_file)
+    return tempdir / slug
+
+
+def _get_full_context(cwd):
+    context_path = cwd.joinpath('.cookiecutter_replay.json')
+    if context_path.exists():
+        with open(context_path) as context_file:
+            context = json.load(context_file)
+    else:
+        raise FileNotFoundError(
+            'Could not find cookiecutter_replay.json, are you in a ballet repo?')
+    with open(PROJECT_CONTEXT_PATH) as context_json:
+        new_context = json.load(context_json)
+    new_keys = set(new_context.keys()) - set(context['cookiecutter'].keys())
+    new_context_config = {'cookiecutter': funcy.project(new_context, new_keys)}
+    new_context = prompt_for_config(new_context_config)
+    context['cookiecutter'].update(new_context)
+    return context
+
+
+def update_project_template():
+    cwd = pathlib.Path(os.getcwd())
+    try:
+        current_repo = git.Repo(
+            safepath(cwd),
+            search_parent_directories=True)  # for right now
+    except BaseException:
+        msg = 'Could not find ballet repo, update failed'
+        logger.exception(msg)
+
+    with tempfile.TemporaryDirectory() as tempdir:
+        updated_template = _create_replay(cwd, tempdir)
+        updated_repo = git.Repo(safepath(updated_template))
+        # add some randomness in the remote name by using tempdir
+        remote_name = updated_template.parts[-1]
+        try:
+            current_repo.heads[TEMPLATE_BRANCH].checkout()
+            updated_remote = current_repo.create_remote(
+                remote_name, updated_repo.working_tree_dir)
+            updated_remote.fetch()
+            current_repo.git.merge(
+                remote_name + '/master',
+                allow_unrelated_histories=True,
+                strategy_option='theirs',
+                squash=True,
+            )
+            current_repo.index.commit(
+                'Merge updates from ballet version {}'.format(version))
+        except BaseException:
+            msg = 'Could not merge changes into template-update branch, update failed'
+            logger.exception(msg)
+        finally:
+            if current_repo.active_branch != current_repo.heads.master:
+                current_repo.heads.master.checkout()
+            current_repo.delete_remote(remote_name)
+
+    try:
+        current_repo.git.merge(
+            TEMPLATE_BRANCH,
+            squash=True,
+        )
+        commit_prompt = 'Would you like ballet to create a merge commit automatically? [y/N]: '
+        answer = input(commit_prompt)
+        if 'y' in answer.lower():
+            current_repo.index.commit(
```

Comment:
I think it makes sense to leave usual merge message, because now the full git history is maintained.
```
master             A --- B -- D
project-template    \ C ---- /
                   ^ "Automatically generated files from ballet-quickstart"
                      ^ "Merge project template updates from ballet v0.5.1"
                         ^ "Update foo"
                              ^ "Merge branch 'project-template'"
```